sbi/inference/base.py

```diff
@@ -58,9 +63,19 @@ def __init__(
         self._show_progressbar = show_progressbar
         self._show_round_summary = show_round_summary
 
-        self._batched_simulator = lambda theta: simulate_in_batches(
-            self._simulator, theta, simulation_batch_size, self._show_progressbar
-        )
+        if num_workers > 1:
```

Comment:
The spirit is to share as much code as possible between the two (so as to minimize the possibility of divergence, and the number of spots where we have to touch stuff), so at the end of the day I am OK with separate functions if commonalities are reduced to a minimum. In particular starting queue/worker for 1 is anyways a matter of adding a conditional, right?

Reply:
I think it would not add a conditional to start just a single worker. So for the sake of minimizing commonalities I agree we should get rid of the if case here. But I am really reluctant to fully rely on multiprocessing when using just a single core, at least as long as we have not tested this extensively on all platforms and are completely convinced about the robustness.

Reply:
OK, I think I haven't been clear (or perhaps the situation is not totally clear in my mind). In any case, I emphatically agree on not relying on mp for a single process scenario.